submit/services/filemanager.py

```diff
@@ -0,0 +1,420 @@
+
+    def request_file(self, path: str, expected_code: int = 200, **kw) \
+            -> Tuple[Download, dict]:
+        """Perform a GET request for a file, and handle any exceptions."""
+        kw.update({'stream': True})
+        resp = self._make_request('get', expected_code, **kw)
+        return Download(resp), resp.headers
+
+    def get_service_status(self) -> dict:
+        """Get the status of the file management service."""
+        return self.request('get', 'status')
+
+    def upload_package(self, pointer: FileStorage) -> Tuple[dict, dict]:
+        """
+        Stream an upload to the file management service.
+
+        If the file is an archive (zip, tar-ball, etc), it will be unpacked.
+        A variety of processing and sanitization routines are performed, and
+        any errors or warnings (including deleted files) will be included in
+        the response body.
+
+        Parameters
+        ----------
+        pointer : :class:`FileStorage`
+            File upload stream from the client.
+
+        Returns
+        -------
+        dict
+            A description of the upload package.
+        dict
+            Response headers.
+        """
+        files = {'file': (pointer.filename, pointer, pointer.mimetype)}
+        return self.request('post', '/', files=files,
+                            expected_code=status.HTTP_201_CREATED)
+
+    def get_upload_status(self, upload_id: str) -> Tuple[dict, dict]:
+        """
+        Retrieve metadata about an accepted and processed upload package.
+
+        Parameters
+        ----------
+        upload_id : str
+            Unique long-lived identifier for the upload.
+
+        Returns
+        -------
+        dict
+            A description of the upload package.
+        dict
+            Response headers.
+
+        """
+        return self.request('get', f'/{upload_id}')
+
+    def add_files_to_package(self, upload_id: str, pointer: FileStorage) \
+            -> Tuple[dict, dict]:
+        """
+        Upload a file or package to an existing upload workspace.
+
+        If the file is an archive (zip, tar-ball, etc), it will be unpacked. A
+        variety of processing and sanitization routines are performed. Existing
+        files will be overwritten by files of the  same name. and any errors or
+        warnings (including deleted files) will be included in the response
+        body.
+
+        Parameters
+        ----------
+        upload_id : str
+            Unique long-lived identifier for the upload.
+        pointer : :class:`FileStorage`
+            File upload stream from the client.
+
+        Returns
+        -------
+        dict
+            A description of the upload package.
+        dict
+            Response headers.
+
+        """
+        return self.request('post', f'/{upload_id}', files={'file': pointer})
+
+    def delete_workspace(self, upload_id: str) -> Tuple[dict, dict]:
+        """
+        Delete the entire workspace.
```

Comment:
Would it make sense to include any comments about authorization to perform this function or conditions under which it might happen? 
Docstrings for create and add are more detailed - perhaps a comment about confirmation back to the UI?